### DISCUSSION TITLE
Cannot launch some games due to DRM issue

### DISCUSSION THREAD

**Discussion by bouncinmachine** [LizardByte/community#266] [Sunshine]
Hi everyone,
First thanks for the impressive work (I am a new Sunshine user)

Some of my games seem to be working fine so far - for example with The Talos Principle (Steam), in the attached logs ;
But for some of them, I get the following error about DRM issue (see screenshot) - for example: The Walking Dead (Steam), in the attached logs. Unfortunately I get this for many of the games I have tried to launch so far.

![error_msg](https://github.com/LizardByte/Sunshine/assets/138763291/8d4a9661-340f-422c-ac5f-827011b34892)
[sunshine_logs.txt](https://github.com/LizardByte/Sunshine/files/11965859/sunshine_logs.txt)

Is there anything I can do about this ?
Many thanks in advance !
Florent

**Comment by atomicmilkshake** [comment]
I'm not an expert, but maybe this will help. I had this issue when I had my Hulu app open with a show I had been watching left paused.  I closed Hulu and restarted my Moonlight session, and it was working again.

**Reply by bouncinmachine** [reply]
Thanks ! Unfortunately I think I have a different problem, as I have no Hulu or similar app open (only the basics : Steam, AMD Adrenalin, etc)

**Reply by pkniestswb** [reply]
In my case it was KeepassXC, so perhabs you have also a password manager app.

**Comment by 13Gears** [comment]
I am having this same issue when I try to launch applications using a shortcut. Every game I try sends this error back client side while the host launches the game just fine. I'm able to play games just fine if I start desktop or steam big picture mode and then launch games from there. I don't have any video/ music streaming apps/programs installed. I checked the FAQ and I don't have any programs in that list either.
![Screenshot_20230708-203253](https://github.com/LizardByte/Sunshine/assets/135494089/23f49b63-bdc9-4ded-af0f-7151a9b3ee40)

**Reply by o10s** [reply]
Same problem with Steam games

**Comment by kieranpx** [comment]
I am having the same issue. I recently switched over to Sunshine for hosting but have not been able to get shortcuts to work without using detached commands. I usually get the same popup about DRM-protected content, even when I'm not running any apps that use DRM.

If I copy the game executable location into the detached command field instead of the command field, then I can launch without issue. The downside (which is expected with detached commands) is that the game does not shut down when ending the stream.

Also, out of curiosity I downloaded The Talos Principle, since I already owned it. Just like you, I am able to launch this game using a standard command without any issues.

Can we make this an issue/bug submission at this point?

**Reply by ReenigneArcher** [reply]
> If I copy the game executable location into the detached command field instead of the command field, then I can launch without issue. The downside (which is expected with detached commands) is that the game does not shut down when ending the stream.

This is expected behavior for many games and apps. The original process starts a secondary process (often after an automatic update check or something like that) and then Sunshine cannot track the original process.

So, there is no issue at this point. Sunshine has no control over how every app/game executes their processes.

**Comment by asarkar157** [comment]
I've found that NVidia Gamestream does not give the DRM error for these apps that Sunshine causes the error for. Still not sure why

**Reply by ReenigneArcher** [reply]
They use a different capture method (NVFBC which is depreciated on Windows).

**Reply by cgutman** [reply]
It also uses a different mechanism for tracking running apps which has different pros and cons compared to ours.

**Reply by asarkar157** [reply]
Thank you! I found a solution someone posted to get past this issue: https://github.com/foxy82/sunshine_utils
Here it is for anyone else who got stuck on this. It was not DRM-related at all, but rather that Sunshine wasn't able to track the Playnite process correctly when initiating Playnite via gamestream via batch script. Totally understandable!

**Reply by bouncinmachine** [reply]
Thanks @asarkar157 But is the link correct ? This one is about the res change script (which does not seem related to the issue) ; also I don't use Playnite.
Are you suggesting to use Playnite + sunshine_utils as a workaround ?
Sorry if there's something I don't get here ;)

[a comment by an automated account is not shown]
